### PR TITLE
Add `notebook-shim`

### PR DESCRIPTION
**Pull request by jtpio**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/notebook-shim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+about:
+  home: https://github.com/jupyterlab/notebook_shim
+  summary: A shim layer for notebook traits and config
+  license: BSD-3-Clause
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - jtpio
+    - Zsailer
```
Comment (jtpio) [inline recipes/notebook-shim/meta.yaml:43]: cc @Zsailer if you are fine being listed here. Thanks!

**Comment by bollwyvl** [comment]
I'm happy to help maintain as well

**Comment by jtpio** [comment]
Thanks @bollwyvl!

File: recipes/notebook-shim/meta.yaml (hunk 1 above)
Comment (Zsailer) [reply recipes/notebook-shim/meta.yaml:44]: :+1:

**Comment by jtpio** [comment]
@conda-forge/help-python this is ready for review, thanks!

**Comment by bollwyvl** [comment]
and, for visibility: @conda-forge/staged-recipes: this  blocks https://github.com/conda-forge/nbclassic-feedstock/pull/19 :ghost:

**Comment by bollwyvl** [comment]
Thank you!

**Pull request merged**